plugin/pkg/shoot/validator/admission.go

```diff
@@ -215,6 +215,11 @@ func (v *ValidateShoot) Admit(ctx context.Context, a admission.Attributes, o adm
 		if reflect.DeepEqual(shoot.Spec, oldShoot.Spec) && reflect.DeepEqual(shoot.ObjectMeta, oldShoot.ObjectMeta) {
 			return nil
 		}
+
+		if (oldShoot.Spec.SeedName == nil && shoot.Spec.SeedName != nil) ||
```

Comment:
Let's also add a case here for `oldShoot.Spec.SeedName != nil && shoot.SpecSeedName == nil`

Reply:
Done. @72b8529

Reply:
Sorry, now that I look at this again can't we just replace it with `reflect.DeepEqual()`?